sdk/eventhub/event-hubs/src/eventHubBufferedProducerClient.ts

```diff
@@ -0,0 +1,315 @@
+  /**
+   * When `true`, all buffered events that are pending should be sent before closing.
+   * When `false`, abandon all buffered events and close immediately.
+   * Defaults to `false`.
+   */
+  flush?: boolean;
+}
+
+/**
+ * Options to configure the `enqueueEvents` method on the `EventHubBufferedProcuerClient`.
+ */
+export interface EnqueueEventOptions extends SendBatchOptions {}
+
+/**
+ * The `EventHubBufferedProducerClient`is used to publish events to a specific Event Hub.
+ *
+ * Depending on the options specified when events are enqueued, they may be
+ * automatically assigned to a partition, grouped according to the specified partition key,
+ * or assigned a specifically requested partition.
+ *
+ * The `EventHubBufferedProducerClient` does not publish events immediately.
+ * Instead, events are buffered so they can be efficiently batched and published
+ * when the batch is full or the `maxWaitTimeInMs` has elapsed with no new events
+ * enqueued.
+ */
+export class EventHubBufferedProducerClient {
+  /**
+   * Describes the amqp connection context for the client.
+   */
+  private _context: ConnectionContext;
+
+  /**
+   * The options passed by the user when creating the EventHubClient instance.
+   */
+  private _clientOptions: EventHubClientOptions;
+
+  /**
+   * @readonly
+   * The name of the Event Hub instance for which this client is created.
+   */
+  get eventHubName(): string {
+    throw new Error("Not implemented");
+  }
+
+  /**
+   * @readonly
+   * The fully qualified namespace of the Event Hub instance for which this client is created.
+   * This is likely to be similar to <yournamespace>.servicebus.windows.net.
+   */
+  get fullyQualifiedNamespace(): string {
+    throw new Error("Not implemented");
+  }
+
+  /**
+   * The `EventHubBufferedProducerClient` class is used to send events to an Event Hub.
+   * Use the `options` parmeter to configure retry policy or proxy settings.
+   * @param connectionString - The connection string to use for connecting to the Event Hub instance.
+   * It is expected that the shared key properties and the Event Hub path are contained in this connection string.
+   * e.g. 'Endpoint=sb://my-servicebus-namespace.servicebus.windows.net/;SharedAccessKeyName=my-SA-name;SharedAccessKey=my-SA-key;EntityPath=my-event-hub-name'.
+   * @param options - A set of options to apply when configuring the client.
+   * - `retryOptions`   : Configures the retry policy for all the operations on the client.
+   * For example, `{ "maxRetries": 4 }` or `{ "maxRetries": 4, "retryDelayInMs": 30000 }`.
+   * - `webSocketOptions`: Configures the channelling of the AMQP connection over Web Sockets.
+   * - `userAgent`      : A string to append to the built in user agent string that is passed to the service.
+   */
+  constructor(connectionString: string, options: EventHubBufferedProducerClientOptions); // eslint-disable-line @azure/azure-sdk/ts-naming-options
```

Comment:
I am surprised `@azure/azure-sdk/ts-naming-options` needs to be disabled here. What did it expect instead?

Reply:
I had copy pasted this from the EventHubProducerClient which had EventHubClientOptions as the `options` type. I'm able to remove this disable comment now.